runners/flink/1.10/src/main/java/org/apache/beam/runners/flink/FlinkCapabilities.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.runners.flink;
+
+/** Handle different capabilities between flink versions. */
+public class FlinkCapabilities {
+
+  /**
+   * Support for outputting elements in close method of chained drivers.
+   *
+   * <p>{@see <a href="https://issues.apache.org/jira/browse/FLINK-14709" FLINK-14709</a>} for more
+   * details.
+   *
+   * @return True if feature is supported.
+   */
+  public static boolean supportsOutputInTearDown() {
```

Comment:
Slightly confusing name because it outputs during close, not teardown; the two are distinct lifecycle states.
```suggestion
  public static boolean supportsOutputDuringClosing() {
```

Reply:
Granted, there is no teardown method in batch. Still, I think we should this clear.

Reply:
👍